pkg/services/blockfetcher/blockfetcher.go

```diff
@@ -262,7 +265,11 @@ func (bfs *Service) blockDownloader() {
 		case <-bfs.ctx.Done():
 			return
 		default:
-			err = bfs.enqueueBlock(b)
+			if !bfs.cfg.BlocksOnly {
+				err = bfs.enqueueHeader(&b.Header)
```

Comment:
We need the code that will read header from stream without transactions reading.